### PR TITLE
add tests for stroke-dasharray with lty

### PR DESCRIPTION
**Pull request by timelyportfolio**
add test for `stroke-dasharray` for `lty=1:6`

### CODE REVIEW

File: tests/testthat/test-lines.R [hunk 1]
```diff
@@ -16,3 +16,36 @@ test_that("lines don't have fill", {
   })
   expect_equal(xml_attr(xml_find_one(x, ".//polyline"), "fill"), "none")
 })
+
+
+test_that("lines lty becomes stroke-dasharray", {
+  x <- xmlSVG({ plot(1:3,lty=1,type="l") })
```
Comment (hadley) [inline tests/testthat/test-lines.R]: Could you remove some of the duplication by making a function that retrieves the dasharray with a function

File: tests/testthat/test-lines.R [hunk 2]
```diff
@@ -16,3 +16,36 @@ test_that("lines don't have fill", {
...
+  expect_true(
```
Comment (hadley) [inline tests/testthat/test-lines.R]: `expect_equal(..., NA_character_)`  would be a better test

File: tests/testthat/test-lines.R [hunk 3]
```diff
@@ -16,3 +16,36 @@ test_that("lines don't have fill", {
...
+    is.na(xml_attr(suppressWarnings(xml_find_one(x,"//polyline")), "stroke-dasharray"))
+  )
+  x <- xmlSVG({ plot(1:3,lty=2,type="l") })
```
Comment (hadley) [inline tests/testthat/test-lines.R]: The `{}` are redundant, and please add spaces around `=`

File: tests/testthat/test-lines.R (hunk 1 above)
Comment (timelyportfolio) [inline tests/testthat/test-lines.R]: Sure I will clean it up.  Is testing grid based systems redundant or should I add those also?

**Comment by timelyportfolio** [comment]
@hadley,  hope newest pull addresses notes and adds `ggplot2` + `lattice`

File: tests/testthat/test-lines.R [hunk 4]
```diff
@@ -16,3 +18,53 @@ test_that("lines don't have fill", {
...
+      2:6,
+      function( l ) {
+        x <- xmlSVG( plot( 1:3, lty = l, type = "l" ) )
+        xml_attr(suppressWarnings(xml_find_one(x,"//polyline")), "stroke-dasharray")
+      }
+    ),
+    stroke_dash
+  )
+
+  # lattice
```
Comment (hadley) [inline tests/testthat/test-lines.R]: Please remove lattice and ggplot2 examples - they don't add anything. The graphics device sits underneath grid, so no need to test for low-level stuff like lty

File: tests/testthat/test-lines.R [hunk 5]
```diff
@@ -16,3 +18,53 @@ test_that("lines don't have fill", {
...
+
+  # base graphics
+  x <- xmlSVG( plot( 1:3, lty = 1, type = "l" ) )
+  expect_equal(
+    xml_attr(suppressWarnings(xml_find_one(x,"//polyline")), "stroke-dasharray"),
+    NA_character_
+  )
+
+  expect_identical(
+    sapply(
```
Comment (hadley) [inline tests/testthat/test-lines.R]: This formulation puts the expected and actual values very far apart. I'd write it like:

```R
dash_array <- function(lty) {
   x <- xmlSVG(plot(1:3, lty = lty, type = "l"))
   xml_attr(xml_find_one(x, "//polyline"), "stroke-dasharray")
}
expect_equal(dash_array(1), "5,5")
```

etc

File: tests/testthat/test-lines.R [hunk 6]
```diff
@@ -16,3 +18,53 @@ test_that("lines don't have fill", {
...
+      2:6,
+      function( l ) {
```
Comment (hadley) [inline tests/testthat/test-lines.R]: Now you have too many spaces - please don't put them inside `()`. You might want to read http://r-pkgs.had.co.nz/r.html#style

**Comment by timelyportfolio** [comment]
ok, styling should be fixed and lattice/ggplot2 removed.

File: tests/testthat/test-lines.R [hunk 7]
```diff
@@ -1,6 +1,8 @@
 context("Lines")
 
 library(xml2)
+library(ggplot2)
```
Comment (hadley) [inline tests/testthat/test-lines.R]: No longer needed

File: tests/testthat/test-lines.R [hunk 8]
```diff
@@ -16,3 +17,23 @@ test_that("lines don't have fill", {
   })
   expect_equal(xml_attr(xml_find_one(x, ".//polyline"), "fill"), "none")
 })
+
+
+test_that("lines lty becomes stroke-dasharray", {
+  # base graphics
+  x <- xmlSVG(plot(1:3, lty = 1, type = "l"))
```
Comment (hadley) [inline tests/testthat/test-lines.R]: `expect_equal(dash_array(1), NA_character_)` ?

File: tests/testthat/test-lines.R [hunk 9]
```diff
@@ -16,3 +17,23 @@ test_that("lines don't have fill", {
...
+  expect_equal(
+    xml_attr(suppressWarnings(xml_find_one(x,"//polyline")), "stroke-dasharray"),
+    NA_character_
+  )
+
+  dash_array <- function(lty) {
+    x <- xmlSVG(plot(1:3, lty = lty, type = "l"))
+    xml_attr(suppressWarnings(xml_find_one(x, "//polyline")), "stroke-dasharray")
```
Comment (hadley) [inline tests/testthat/test-lines.R]: Why do you have `suppressWarnings()` here?

Comment (timelyportfolio) [inline tests/testthat/test-lines.R]: more than one polyline so `xml_find_one` returns warning.  I'll change to use miniplot.

**Comment by hadley** [comment]
Reading "line type specification" in `?par` I'm not actually sure these values are correct:

> The five standard dash-dot line types (lty = 2:6) correspond to c("44", "13", "1343", "73", "2262").

**Comment by timelyportfolio** [comment]
for reference, here are the [lines](https://github.com/sjp/gridSVG/blob/de85b6dd0b43325a60cb675a583e15346dfaf16e/pkg/R/devsvg.R#L50-L59) from `gridSVG`.

**Comment by hadley** [comment]
Thanks. I finished it off myself so I can switch to another project for the rest of today.  Please keep the bug reports and pull requests coming!

**Comment by timelyportfolio** [comment]
sure.  Have started a bunch of complicated tests on my end for all the various graphics libraries, so will report back.  I'm so excited about this finally getting done, so happy to help in any way possible.  Once stable, I'll try to think up some vignettes.

```
mini_plot(sin(seq.int(0,10,by=0.02)),type="l",lty=2,lwd=4,col=blues9[8])
```

image

**Comment by hadley** [comment]
FWIW I think it's useful to test complicated graphics to see what fails, but you should be able to reduce any tests for this repo down to just a line or two of base graphics (or maybe grid, but in most cases it shouldn't be necessary)

**Comment by hadley** [comment]
Also raster plots and plotmath don't work, and won't work until @davidgohel has some time to put into it.

**Pull request merged**